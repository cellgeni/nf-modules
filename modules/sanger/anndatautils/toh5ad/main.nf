

process ANNDATAUTILS_TOH5AD {
    tag "Converting ${sample_id}'s file to .h5ad"
    input:
        tuple val(sample_id), path(input, name: 'input/*')
        val(delimiter)
    output:
        tuple val(sample_id), path("${sample_id}.h5ad")
    script:
        """
        convert_to_h5ad.py \
            --input ${input} \
            --sample_id ${sample_id} \
            --delimiter ${delimiter} \
            --output ${sample_id}.h5ad
        """
}
