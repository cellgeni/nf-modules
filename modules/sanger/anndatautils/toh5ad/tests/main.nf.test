nextflow_process {

    name "Test Process ANNDATAUTILS_TOH5AD"
    script "../main.nf"
    process "ANNDATAUTILS_TOH5AD"

    tag "modules"
    tag "modules_sanger"
    tag "anndatautils"
    tag "anndatautils/toh5ad"

    // TODO nf-core: Change the test name preferably indicating the test-data and file-format used
    test("Converts .h5 file to .h5ad") {
        when {
            process {
                """
                input[0] = [
                    ['pbmc_1k_h5', file("./data/anndatautils/pbmc_1k.h5"],
                    ['pbmc_1k_mtx', file("./data/anndatautils/raw_feature_bc_matrix"]
                ]
                input[1] = ["___", "---"]
                """
            }
        }

        then {
            // basec run health
            assert process.success
            assert process.exitStatus == 0
            assert process.trace.succeeded().size() == 1

            // stdout and stderr hygiene
            assert process.stdout.any { it.contains('Saving AnnData object to .h5ad file') }
            assert process.stderr.isEmpty()

            // output files
            with(process.out.h5ad) {
                assert size() == 2
                each { f -> assert path(f).exists() } 
                
                def names = collect { path(it).name }
                assert names == [
                    'pbmc_1k_h5.h5ad',
                    'pbmc_1k_mtx.h5ad'
                ]
            }

            with(process.out.versions) {
                assert size() == 2
                each { f -> assert path(f).exists() } 
            }

        }
        
    }

}
