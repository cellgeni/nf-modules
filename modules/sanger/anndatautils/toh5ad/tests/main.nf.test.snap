{
    "Conversion of 10x .mtx file to .h5ad": {
        "content": [
            {
                "0": [
                    [
                        "pbmc_1k_mtx",
                        "pbmc_1k_mtx.h5ad:md5,767be43af6204bc0069a74ea8dc7f510"
                    ]
                ],
                "1": [
                    "versions.yml:md5,8e108526f7232315b87c55ec4b8d7887"
                ],
                "h5ad": [
                    [
                        "pbmc_1k_mtx",
                        "pbmc_1k_mtx.h5ad:md5,767be43af6204bc0069a74ea8dc7f510"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,8e108526f7232315b87c55ec4b8d7887"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-05-02T15:03:33.624797"
    },
    "Run -stub for .mtx and .h5 files": {
        "content": [
            {
                "0": [
                    [
                        "pbmc_1k_h5",
                        "pbmc_1k_h5.h5ad:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ],
                    [
                        "pbmc_1k_mtx",
                        "pbmc_1k_mtx.h5ad:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "versions.yml:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "h5ad": [
                    [
                        "pbmc_1k_h5",
                        "pbmc_1k_h5.h5ad:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ],
                    [
                        "pbmc_1k_mtx",
                        "pbmc_1k_mtx.h5ad:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "versions.yml:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-05-02T15:03:57.025663"
    },
    "Conversion of .h5 file to .h5ad": {
        "content": [
            {
                "0": [
                    [
                        "pbmc_1k_h5",
                        "pbmc_1k_h5.h5ad:md5,dd0b6bedb3ccab368da6fbc97d71ac3f"
                    ]
                ],
                "1": [
                    "versions.yml:md5,8e108526f7232315b87c55ec4b8d7887"
                ],
                "h5ad": [
                    [
                        "pbmc_1k_h5",
                        "pbmc_1k_h5.h5ad:md5,dd0b6bedb3ccab368da6fbc97d71ac3f"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,8e108526f7232315b87c55ec4b8d7887"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-05-02T15:02:58.33419"
    }
}